src/main/java/appeng/parts/p2p/PartP2PFluids.java

```diff
@@ -234,79 +293,14 @@ private IFluidHandler getTarget()
 		}
 
 		final TileEntity te = this.getTile().getWorld().getTileEntity( this.getTile().getPos().offset( this.getSide().getFacing() ) );
-		if( te instanceof IFluidHandler )
-		{
-			return this.cachedTank = (IFluidHandler) te;
-		}
-
-		return null;
-	}
 
-	@Override
-	public FluidStack drain( final EnumFacing from, final FluidStack resource, final boolean doDrain )
-	{
-		return null;
-	}
-
-	@Override
-	public FluidStack drain( final EnumFacing from, final int maxDrain, final boolean doDrain )
-	{
-		return null;
-	}
-
-	@Override
-	public boolean canFill( final EnumFacing from, final Fluid fluid )
-	{
-		return !this.isOutput() && from == this.getSide().getFacing() && !this.getOutputs( fluid ).isEmpty();
-	}
-
-	@Override
-	public boolean canDrain( final EnumFacing from, final Fluid fluid )
-	{
-		return false;
-	}
-
-	@Override
-	public FluidTankInfo[] getTankInfo( final EnumFacing from )
-	{
-		if( from == this.getSide().getFacing() )
-		{
-			return this.getTank();
-		}
-		return new FluidTankInfo[0];
-	}
-
-	private FluidTankInfo[] getTank()
-	{
-		if( this.isOutput() )
-		{
-			final PartP2PLiquids tun = this.getInput();
-			if( tun != null )
-			{
-				return ACTIVE_TANK;
-			}
-		}
-		else
+		if( te != null && te.hasCapability( CapabilityFluidHandler.FLUID_HANDLER_CAPABILITY, this.getSide().getFacing().getOpposite() ) )
 		{
-			try
-			{
-				if( !this.getOutputs().isEmpty() )
-				{
-					return ACTIVE_TANK;
-				}
-			}
-			catch( final GridAccessException e )
-			{
-				// :(
-			}
+			return this.cachedTank = te.getCapability( CapabilityFluidHandler.FLUID_HANDLER_CAPABILITY,
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Extract the assignment out of this expression. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AAssignmentInSubExpressionCheck)